specification/storage/resource-manager/Microsoft.Storage/stable/2017-10-01/storage.json

```diff
@@ -974,7 +974,18 @@
         "services": {
           "$ref": "#/definitions/EncryptionServices",
           "description": "List of services which support encryption."
-        },
+        }    
+      },
+      "allOf":  [       
+        {        
+          "$ref": "#/definitions/EncryptionKeySource"          
+        }    
+      ],
+      "required": [ "keySource" ],
```

Comment:
Adding a new required property is a breaking change. This is not allowed for stable Swagger specs.

Reply:
@jianghaolu FYI, I was looking at the Python code, and Github diff is misleading in this case, this was already a required property.
However, `Encrypton` is now a sub-class of `EncryptionKeySource` (new class). That shuffles model signatures a bit. Also some models now are supposed to return an `EncryptionKeySource` and not `Encryption`. This one is definitely breaking, but I can't judge if it's relevant or not.

Reply:
@jianghaolu，"keySource" is required in the original input type #/definitions/Encryption, so this is not breaking.

@lmazuel, I can remove the required from 'Encryptin' (this type only use in output now). We use ‘Encrypton’ for both input and output before, but now only in output, and use "EncryptionKeySource" in input. This is because we should not input Encryption Service anymore, but only Keysource. The change only in request input, but not output. If change the input is a breaking (I think it should be), I will contact feature team to see how to handle it.

Reply:
Thanks @lmazuel for taking an extra look -
Yes it's a breaking change to remove optional parameters or properties because old code that set the property will be broken now.